File: poo/java/Payroll/SalariedEmployee.java
import java.util.Date;

public class SalariedEmployee extends Employee{

    /* Atributos */
    private double weeklySalary;

    /* Construtor */
    public SalariedEmployee(String first, String last, String ssn, 
                            Date birthDate, int departmentCode,
                            double salary){

        super(first, last, ssn, birthDate, departmentCode); 
        this.setWeeklySalary(salary); 

    }

    /* Getters */
    public double getWeeklySalary(){
        return this.weeklySalary;
    }

    /* Setters */
    public void setWeeklySalary(double salary){
        this.weeklySalary = salary < 0.0 ? 0.0 : salary;
    }

    /* Calcula os ganhos do funcionario */
    public double earnings()                                            
    {       
        Date today = new Date();
        if(today.getMonth() == super.getPrivateBirthDate().getMonth())
            return this.getWeeklySalary()+150;
        return this.getWeeklySalary();
    }              

    /* Metodo invocado quando tentamos imprimir objetos desta classe */
    public String toString()                                     
    {                                                            
        return String.format( "salaried employee: %s\n%s: $%,.2f",
            super.toString(), "weekly salary", getWeeklySalary() );
    }
}

File: poo/python/Payroll/SalariedEmployee.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 14/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from datetime import date
from Employee import Employee

class SalariedEmployee(Employee):

    # Atributos #
    __weeklySalary = 0

    # Construtor #
    def __init__(self, first, last, ssn, birthDate, departmentCode, salary):
        super(type(self), self).__init__(first, last, ssn,
                                               birthDate, departmentCode)
        self.setWeeklySalary(salary)

    # Getters #
    def getWeeklySalary(self):
        return self.__weeklySalary

    # Setters #
    def setWeeklySalary(self, weeklySalary):
        self.__weeklySalary = weeklySalary

    # Calcula os ganhos do funcionario #
    def earnings(self):
        hj = date.today()
        pai = super(type(self), self)
        if hj.month == pai.getPrivateBirthDate().month:
            return self.getWeeklySalary() + 150
        return self.getWeeklySalary()

    # Representação textual dos objetos #
    def __str__(self):
        pai = super(type(self), self)
        return '''salaried employee: {0}
                  weekly salary: {1}'''.format(pai.__str__(), 
                                               self.getWeeklySalary())

    def __repr__(self):
        return self.__str__()
